management/server/grpcserver.go

```diff
@@ -109,17 +112,52 @@ func (s *GRPCServer) GetServerKey(ctx context.Context, req *proto.Empty) (*proto
 	}, nil
 }
 
+// extractIps returns real peer ip and grpc peer address (ip:port)
+func extractIps(ctx context.Context) (string, string) {
+	var peerAddr string
+	grpcPeer, grcpPeerExtracted := gRPCPeer.FromContext(ctx)
+	if grcpPeerExtracted {
+		peerAddr = grpcPeer.Addr.String()
+	}
+
+	if headers, ok := metadata.FromIncomingContext(ctx); ok {
+		value := headers.Get("x-forwarded-for")
+		if len(value) == 0 || value[0] == "" {
+			value = headers.Get("x-real-ip")
+		}
+
+		if len(value) > 0 && value[0] != "" {
+			ips := strings.Split(value[0], ",")
+			ipStr := strings.TrimSpace(ips[len(ips)-1])
+			ip, err := netip.ParseAddr(ipStr)
+			if err == nil {
+				return ip.String(), peerAddr
+			}
+		}
+
+		if grcpPeerExtracted {
+			ip, err := net.ResolveTCPAddr("", grpcPeer.Addr.String())
```

Comment:
Using netip.ParseAddrPort() might be better for this case as it works with the assumption of IP address x the domain and protocol checks from net.ResolveTCPAddr

Reply:
This is much better. Thank you!